lib/rules/jsx-max-depth.js

```diff
@@ -90,7 +90,8 @@ module.exports = {
 
             return isJSXElement(writeExpr)
               && writeExpr
-              || writeExpr.type === 'Identifier'
+              || writeExpr
```

Comment:
probably this whole chain should be rewritten for clarity; at the least, let's use parens so as to not mix && and || :-)